db/schema_tables.cc

```diff
@@ -2222,11 +2222,54 @@ std::vector<mutation> make_drop_aggregate_mutations(schema_features features, sh
  * Table metadata serialization/deserialization.
  */
 
+/// Returns mutations which when applied to the database will cause all schema changes
+/// which create or alter the table with a given name, made with timestamps smaller than t,
+/// have no effect. Used when overriding schema to shadow concurrent conflicting schema changes.
+/// Shouldn't be needed if schema changes are serialized with RAFT.
+static schema_mutations make_table_deleting_mutations(const sstring& ks, const sstring& table, bool is_view, api::timestamp_type t) {
+    tombstone tomb;
+
+    // Generate neutral mutations if t == api::min_timestamp
+    if (t > api::min_timestamp) {
+        tomb = tombstone(t - 1, gc_clock::now());
+    }
+
+    auto tables_m_s = is_view ? views() : tables();
+    mutation tables_m{tables_m_s, partition_key::from_singular(*tables_m_s, ks)};
+    {
+        auto ckey = clustering_key::from_singular(*tables_m_s, table);
+        tables_m.partition().apply_delete(*tables_m_s, ckey, tomb);
+    }
+
+    mutation scylla_tables_m{scylla_tables(), partition_key::from_singular(*scylla_tables(), ks)};
+    {
+        auto ckey = clustering_key::from_singular(*scylla_tables(), table);
+        scylla_tables_m.partition().apply_delete(*scylla_tables(), ckey, tomb);
+    }
+
+    auto make_drop_columns = [&] (const schema_ptr& s) {
+        mutation m{s, partition_key::from_singular(*s, ks)};
+        auto ckey = clustering_key::from_exploded(*s, {utf8_type->decompose(table)});
+        m.partition().apply_delete(*s, ckey, tomb);
+        return m;
+    };
+
+    return schema_mutations(std::move(tables_m),
+                            make_drop_columns(columns()),
+                            make_drop_columns(view_virtual_columns()),
+                            make_drop_columns(computed_columns()),
+                            mutation(indexes(), partition_key::from_singular(*indexes(), ks)),
```

Comment:
I thought I asked this but I don't see my comment: why isn't this a tombstone?

Reply:
Well, it could be a tombstone, but it probably won't help because indexes are created with higher timestamps anyway, because they are created after "create table" succeeds. And we don't want to drop later indexes.